core/src/main/java/org/apache/cxf/common/jaxb/JAXBUtils.java

```diff
@@ -1346,7 +1346,11 @@ private static Object createEclipseNamespaceMapper(Class<?> mcls, Map<String, St
             + ("RI".equals(postFix) ? "" : "internal/")
             + "bind/marshaller/NamespacePrefixMapper";
         String postFixedName = "org/apache/cxf/jaxb/NamespaceMapper" + postFix;
+        Class<?> cls = helper.findClass(className, ref);
```

Comment:
Same as for `createFactory`, it would be better to distinguish the case when suitable factory exists vs creating a new one.